File: soluciones/java/Ejercicio09.java
/*
 * Ejercicio 09 - Nivel Intermedio
 * Llenar dos vectores A y B de 5 elementos, sumar elemento a elemento
 * (C[i] = A[i] + B[i]) e imprimir el vector C.
 */
import java.util.Scanner;

public class Ejercicio09 {

    static void sumarVectores(int[] A, int[] B, int[] C, int tam) {
        for (int i = 0; i < tam; i++) {
            C[i] = A[i] + B[i];
        }
        System.out.println("Vector C (resultado A + B):");
        for (int i = 0; i < tam; i++) {
            System.out.println("C[" + i + "] = " + C[i]);
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int tam = 5;
        int[] A = new int[tam];
        int[] B = new int[tam];
        int[] C = new int[tam];
        for (int i = 0; i < tam; i++) {
            System.out.print("Ingrese el elemento " + i + " de A: ");
            A[i] = sc.nextInt();
            System.out.print("Ingrese el elemento " + i + " de B: ");
            B[i] = sc.nextInt();
        }
        sumarVectores(A, B, C, tam);
        sc.close();
    }
}

File: soluciones/cpp/ejercicio09.cpp
/*
 * Ejercicio 09 - Nivel Intermedio
 * Llenar dos vectores A y B de 5 elementos, sumar elemento a elemento
 * (C[i] = A[i] + B[i]) e imprimir el vector C.
 */
#include <iostream>
using namespace std;

void sumarVectores(int A[], int B[], int C[], int tam) {
    for (int i = 0; i < tam; i++) {
        C[i] = A[i] + B[i];
    }
    cout << "Vector C (resultado A + B):" << endl;
    for (int i = 0; i < tam; i++) {
        cout << "C[" << i << "] = " << C[i] << endl;
    }
}

int main() {
    int tam = 5;
    int* A = new int[tam];
    int* B = new int[tam];
    int* C = new int[tam];
    for (int i = 0; i < tam; i++) {
        cout << "Ingrese el elemento " << i << " de A: ";
        cin >> A[i];
        cout << "Ingrese el elemento " << i << " de B: ";
        cin >> B[i];
    }
    sumarVectores(A, B, C, tam);

    delete[] A;
    delete[] B;
    delete[] C;
    return 0;
}
